### PR TITLE
Add cognitive service AnomalyDetector SDK with unittests

### PR DESCRIPTION
**Pull request by moreOver0**
https://github.com/Azure/azure-rest-api-specs/pull/5295 This is the related REST spec API PR.

### CODE REVIEW

**Comment by moreOver0** [comment]
@yangyuan Please take a reivew except the file cognitiveservices_data-plane_AnomalyDetector.txt because it's not real log when generating code. I will update this file when Azure/azure-rest-api-specs#5259 get merged.

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Properties/AssemblyInfo.cs
Comment (yangyuan) [inline src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Properties/AssemblyInfo.cs]: any reason use 2.* instead of 1.* or 0.*?

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj
Comment (yangyuan) [inline src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj]: You want directly publish a public release?
As it's first version, I personally think publishing a preview SDK first is a better idea?

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Properties/AssemblyInfo.cs
Comment (moreOver0) [reply src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Properties/AssemblyInfo.cs]: I thought this should align with the version in swagger file.

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj
Comment (moreOver0) [reply src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj]: then make this to be 1.0.0.0

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Properties/AssemblyInfo.cs
Comment (moreOver0) [reply src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Properties/AssemblyInfo.cs]: then make this to be 1.0.0.0

**Review by yangyuan — APPROVED** [review]

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj]: If you intend to start with a preview release, please start the versioning from `0.8.0-preview`

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj]: Please remove `Description`

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Properties/AssemblyInfo.cs
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Properties/AssemblyInfo.cs]: Please update the `AssemblyInfo`

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj
Comment (moreOver0) [reply src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj]: But why?

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj
Comment (moreOver0) [reply src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj]: If there some reason to do so? I have not seen any document/rule about version. You can give me some link if it exists so that I can go through the doc.

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj
Comment (dsgouda) [reply src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj]: The description for nuget package should be described in the `PackageReleaseNotes` section per our guidelines, since you already have that in the csproj, `Description` is not necessary

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj
Comment (dsgouda) [reply src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj]: This is the link for semantic versioning in general. For brand new preview Nuget packages, we have a guideline for versioning the packages starting (0.8.0-preview)

**Comment by dsgouda** [comment]
@moreOver0 please squash the commits for a cleaner commit history

**Review by dsgouda — COMMENTED** [review]
Approved in principle. Will approve and merge once the code is generated from merged REST spec

**Review by shahabhijeet — APPROVED** [review]

**Comment by shahabhijeet** [comment]
@AlexGhiondea can you approve this as this is dataplane PR.

**Review by AlexGhiondea — APPROVED** [review]

File: src/SDKs/_metadata/cognitiveservices_data-plane_AnomalyDetector.txt
Comment (AlexGhiondea) [inline src/SDKs/_metadata/cognitiveservices_data-plane_AnomalyDetector.txt]: Was this supposed to be added?

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj
Comment (AlexGhiondea) [inline src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj]: I see you are bumping down the version number and changing the name. Any reason to pick 0.8.0 here instead of something else (like 0.1.0)?

**Review by AlexGhiondea — COMMENTED** [review]
I have a question about the package version.

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj
Comment (moreOver0) [reply src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj]: @shahabhijeet merged some project structures in https://github.com/Azure/azure-sdk-for-net/pull/5464 because he didn't aware that there already has a ready SDK PR. So I solved the conflicts after that. Please ignore all these changes.

@dsgouda suggested me to use 0.8.0-preivew in previous comments.

File: src/SDKs/_metadata/cognitiveservices_data-plane_AnomalyDetector.txt
Comment (moreOver0) [reply src/SDKs/_metadata/cognitiveservices_data-plane_AnomalyDetector.txt]: Do you mean this line `cmd.exe /c autorest.cmd https://github.com/moreOver0/azure-rest-api-specs/blob/zhuxia/ad/specification/cognitiveservices/data-plane/AnomalyDetector/readme.md --csharp --version=latest --reflect-api-versions --csharp-sdks-folder=D:\src\azure-sdk-for-net\src\SDKs` ?

I generated the C# code from my own branch because my swagger PR is not merged now. I will update this after the swagger PR get merged.

Comment (AlexGhiondea) [reply src/SDKs/_metadata/cognitiveservices_data-plane_AnomalyDetector.txt]: Sorry -- I was asking about the entire text file.

Comment (moreOver0) [reply src/SDKs/_metadata/cognitiveservices_data-plane_AnomalyDetector.txt]: I thought I should add this file. It is a auto generated log file when I use generate.ps1 to generate code. I believe this file is target to record repo, branch and commit id.

@dsgouda What do you think about this file? Should I add this file?

Comment (dsgouda) [reply src/SDKs/_metadata/cognitiveservices_data-plane_AnomalyDetector.txt]: @AlexGhiondea This is a log file we generate when generating the SDK using scripts. Helps us review what specs were used for it.

File: src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj
Comment (dsgouda) [reply src/SDKs/CognitiveServices/dataPlane/AnomalyDetector/AnomalyDetector/Microsoft.Azure.CognitiveServices.AnomalyDetector.csproj]: @AlexGhiondea For brand new RPs we recommend starting from `0.8.0-preview` so that when they are ready to release a stable version they can start from `1.0.0`. we apply the same guidelines when working with the next preview versions (`1.8.0-preview`-`2.0.0`)

**Comment by dsgouda** [comment]
@AlexGhiondea  I have answered the questions, let us know if this good to merge

**Comment by moreOver0** [comment]
@dsgouda I have updated the generate.ps1 and session records since the swagger PR got merged. You can merge this PR when you and Alex think it's OK.

**Comment by AlexGhiondea** [comment]
LGTM!

**Review by dsgouda — APPROVED** [review]

**Pull request merged**
